elf_ctx.c

```diff
@@ -94,9 +94,25 @@ extern char *vdso_image_addr;
 
 #if CONFIG_ARCH_X86_64
 	static const char *auxv_platform = "x86_64";
+#elif CONFIG_ARCH_ARM_64
+	static const char *auxv_platform = "aarch64";
 #else
 #error "Unsupported architecture"
-#endif /* CONFIG_ARCH_X86_64 */
+#endif
+
+#if CONFIG_ARCH_ARM_64
+/* In ARM_64, `ukarch_rstack_push` aligns the stack pointer to 16
+ * after each push to the stack, but this should not be done when
+ * constructing the initial stack.
+ */
+#define ukarch_rstack_push(sp, value)			\
```

Comment:
Hum... I think it would make sense to move this (macro) function to the Unikraft core as part of `libcontext`. I assume elfloader is not the only use case where you want a packed pushing on the stack of arguments. For sure I'd give it a different name so that it is non conflicting, like `ukarch_rstack_push_packed()`. On x86, we could put an alias macro so that we can use the same code in elfloader. Can you move this macro?

Reply:
Sorry, I may not have understood this correctly. I think the code in elfloader can't be the same, because arm64 needs an non-aligned implementation. Is this updated PR and unikraft/unikraft#1096 what you want?

Reply:
Yes, the solution looks with PR#1096 good, thanks!